spikeinterface/core/segmentutils.py

```diff
@@ -264,5 +267,43 @@ def get_unit_spike_train(self, *args, **kwargs):
 def append_sortings(*args, **kwargs):
     return AppendSegmentSorting(*args, **kwargs)
 
-
 append_sortings.__doc__ == AppendSegmentSorting.__doc__
+
+# Q: should this take as input a concatenated recording or just a list of recordings?
+class SplitSegmentSorting(BaseSorting):
+    """Splits a sorting with a single segment to multiple segments
+    based on the given list of recordings (must be in order)
+
+    Parameters
+    ----------
+    parent_sorting : BaseSorting
+        Sorting with a single segment (e.g. from sorting concatenated recording)
+    recording_list : list
+        List of recordings whose lengths will be used to split the sorting
+        into smaller segments
+    """
+    def __init__(self, parent_sorting: BaseSorting, recording_list: List[BaseRecording]):
+        sampling_frequency = parent_sorting.get_sampling_frequency()
```

Comment:
Can you add an assertion that the input sorting is mono segment?

Reply:
Done